### PR TITLE
Display warning when socat cmd is missing

### PR DESCRIPTION
**Pull request by wtrocki**
## Motivation

I have been troubleshooting couple openshift installations done using `oc cluster up` recently and found one common pattern. MacOS users who used `oc cluster up` for the first time almost always experienced problems with reaching running cluster.  Example:

```
curl 127.0.0.1:8443
curl: (7) Failed connect to 127.0.0.1:8442; Connection refused
```

Problem related with missing or improperly installed `socat` command.
Following [documentation](https://github.com/openshift/origin/blob/master/docs/cluster_up_down.md#macos-with-docker-for-mac) is often not enough. Users who install this command using brew may miss some additional libraries and this is really hard to debug and track cause of the issue when looking into logs. Example output:

```
> socat
dyld: Library not loaded: /usr/local/opt/readline/lib/libreadline.7.dylib
  Referenced from: /usr/local/bin/socat
  Reason: image not found
[1]    13015 abort      socat 
```

Problem is not easy to investigate as currently installer provides no useful message back to the user.

Example issue that was caused by problem
https://github.com/openshift/origin/issues/17931

## Proposed solution

Validate if socat is available when executing `oc cluster up` to inform users about possible problems that may occur. Validation will happen only when `PortForwarding` option will be used.

Implementation using non disruptive aproach - server will still start without `socat` and only warning will be displayed.

## Dev notes

Changes will be squashed after review. 
To properly unit test changes I will need to create fake structure for `localcmd` package in order to mock execution of the command. Something like  [wrappers for os/exec](https://github.com/kubernetes/contrib/tree/master/exec-healthz/vendor/k8s.io/kubernetes/pkg/util/exec) in kubernetes dependencies.
I wasn't sure if this is worth doing for such small change.

## Verification
```
brew uninstall socat
oc cluster up | grep "Port forwarding requires socat"
```

### CODE REVIEW

**Comment by bparees** [comment]
/ok-to-test

@csrwng look reasonable to you?  I don't have a mac to test on :)

**Comment by wtrocki** [comment]
Checking why tests are failing.

File: pkg/oc/bootstrap/docker/up.go [hunk 1]
```diff
@@ -726,6 +730,18 @@ func (c *CommonStartConfig) CheckDockerVersion(out io.Writer) error {
 	return nil
 }
 
+// CheckPortForwardingPrerequisites checks that socat is installed when port forwarding is enabled
+// Socat needs to be installed manually on MacOS
+func (c *CommonStartConfig) CheckPortForwardingPrerequisites(out io.Writer) error {
+	err := localcmd.New("socat").Args("-V").Run()
```
Comment (mfojtik) [inline pkg/oc/bootstrap/docker/up.go]: why not use https://golang.org/pkg/os/exec/#LookPath ?

Comment (wtrocki) [reply pkg/oc/bootstrap/docker/up.go]: I wanted to cover the case when users installed that using brew and have it in the path but it was missing some additional packages. It happens randomly on some machines.
```
socat
dyld: Library not loaded: /usr/local/opt/readline/lib/libreadline.7.dylib
  Referenced from: /usr/local/bin/socat
  Reason: image not found
[1]    13015 abort      socat 
```

**Comment by wtrocki** [comment]
I'm trying to replicate problem on local machine, by building oc client and doing regular cluster up. 
It looks like tests are failing because of problems unrelated with this PR, but I'm not sure about this.

**Comment by bparees** [comment]
actually @wtrocki can you squash your commits? thanks

**Comment by wtrocki** [comment]
```
Summarizing 1 Failure:

[Fail] [sig-storage] Projected [It] should update labels on modification [Conformance] [Suite:openshift/conformance/parallel] [Suite:k8s] 
/tmp/openshift/build-rpm-release/rpm/BUILD/origin-3.9.0/_output/local/go/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/test/e2e/common/projected.go:966
```
It looks like this is unrelated issue.

**Comment by bparees** [comment]
yeah it's a known flake.
/retest

**Pull request merged**